game/video_driver.cc

```diff
@@ -94,7 +150,6 @@ Window::Window() {
 	std::clog << "video/info: GL_RENDERER:   " << glGetString(GL_RENDERER) << std::endl;
 	// Extensions would need more complex outputting, otherwise they will break clog.
 	//std::clog << "video/info: GL_EXTENSIONS: " << glGetString(GL_EXTENSIONS) << std::endl;
-	if (epoxy_gl_version() < 33) throw std::runtime_error("OpenGL 3.3 is required but not available");	
 	createShaders();
 	resize();
 	SDL_ShowWindow(screen);
```

Comment:
Window constructor is getting huge. I'd suggest breaking it into smaller units. If you need some of them initialized in Window initlist, you may still use function-try-block to catch any errors within the constructor.

Reply:
That would probably be a better solution in the long-term. Perhaps we should make the default constructor just allocate resources but add a function that actually creates the window and GL context?

Reply:
As you mentioned earlier, that is usually messy. If at all possible, construct and fail in constructor, so that the object may only exist in completely valid state. The same goes for Game and optional Window, which would suddenly cause exception to be thrown when someone tries to use it.

I believe that in this case you can avoid any partial constructor or optional values, unless I misunderstood somehow how this patch works.

Reply:
The problem is the following: 
There are a number of reasons why the Window constructor can fail. However; if the Window is not allocated inside a try block, the exception goes unhandled and the user gets no feedback of what went wrong because the actual error is buried in the stack trace (which is what happened with the issue this PR originally meant to correct)

Options are either doing what I'm doing, doing what I'm proposing or I suppose eventually handling the exception inside the constructor, but I'm not sure whether we prefer adding try/catch blocks into individual classes rather than what I'm doing?